Source/WebKit/WebProcess/cocoa/VideoPresentationManager.mm

```diff
@@ -400,11 +400,10 @@ static FloatRect inlineVideoFrame(HTMLVideoElement& element)
     if (blockMediaLayerRehosting) {
         contextID = videoElement.layerHostingContextID();
         if (!contextID) {
-            m_setupFullscreenHandler = WTFMove(setupFullscreen);
-            videoElement.requestHostingContextID([this, protectedThis = Ref { *this }, videoElement = Ref { videoElement }] (auto contextID) {
-                if (!contextID || !m_setupFullscreenHandler)
+            videoElement.requestHostingContextID([protectedThis = Ref { *this }, videoElement = Ref { videoElement }, setupFullscreenHandler = WTFMove(setupFullscreen)] (auto contextID) {
+                if (!contextID)
                     return;
-                m_setupFullscreenHandler(contextID, FloatSize(videoElement->videoWidth(), videoElement->videoHeight()));
+                setupFullscreenHandler(contextID, FloatSize(videoElement->videoWidth(), videoElement->videoHeight()));
```

Comment:
(note: another much shorter fix would be to do `m_setupFullscreenHandler = nullptr;`  )